spec/mongoid/validatable/uniqueness_spec.rb

```diff
@@ -19,34 +19,42 @@
         context "when persisting to another collection" do
 
           before do
-            Dictionary.with(collection: "dicts").create(name: "websters")
+            Dictionary.with(collection: "dicts") do |klass|
+              klass.create(name: "websters")
+            end
           end
 
           context "when the document is not valid" do
 
             let(:websters) do
-              Dictionary.with(collection: "dicts").new(name: "websters")
+              valid = nil
```

Comment:
Because if you call #valid? on the object outside of the block, the alternative collection will not be used, thus returning true for valid? instead of the expected false.

I need to call #valid? on the object in the block and access the return value as well as the object itself in the spec so that I can check the errors list.